app/subscribe/views.py

```diff
@@ -15,15 +21,44 @@ def subscribe():
             email = form.user_email.data
             phone = form.user_phone.data
 
-            if email or phone:
-                create_subscriber(first_name=first_name,
-                                  last_name=last_name,
-                                  email=email,
-                                  phone=phone)
-                flash(Markup('Thank you for subscribing!'), category='success')
-                return redirect(url_for('subscribe.subscribe'))
-            else:
+            # Verify recaptcha token
+            recaptcha_response = requests.post(
+                url=f'https://www.google.com/recaptcha/api/siteverify?secret={Config.RECAPTCHA_PRIVATE_KEY}\
+                            &response={request.form["g-recaptcha-response"]}').json()
+
+            if not (email or phone):
                 flash(Markup('Please enter an email address or phone number.'), category='warning')
-                return redirect(url_for('subscribe.subscribe'))
+
+            elif recaptcha_response['success'] is False or recaptcha_response['score'] < 0.5:
+                flash(Markup('Recaptcha failed to validate you!'), category='danger')
+
+            else:
+                # Check entered email and phone number
+                verify_email = verify_subscriber(email, phone)
+
+                if verify_email == EMAIL_TAKEN:
+                    flash(Markup('You are already subscribed with that email!'), category='warning')
+                    return render_template('subscribe/subscribe.html', form=form,
+                                           RECAPTCHA_PUBLIC_KEY=Config.RECAPTCHA_PUBLIC_KEY)
+
+                elif verify_email == PHONE_TAKEN:
+                    flash(Markup('You are already subscribed with that phone number!.'), category='warning')
```

Comment:
Change to "The phone number you entered is already subscribed, please use another phone number".

Reply:
done